### PR TITLE
ath79: add D-Link DIR-825-C1 and DIR-835-A1

### PR DESCRIPTION
**Pull request by micmac1**
Hello all,

Here's my port of these two boards from ar71xx to ath79. I tested a lot with the first one, the second one I don't own. But they're pretty much identical.

All LEDs work. Well, technically 5GHz wifi led doesn't work. But that's due to Felix' commit ccab68f2d399d2395a18d1fb3495fe7e048fe054 and I believe the other ath79 targets that use the ath9k gpio controller have the same problem. I'm sure that will be sorted out in due time.

I kept the behavior from ar71xx regarding MAC addresses and LEDs, except that now the blue Power LED is used when running instead of the orange one. That's also how the vendor firmware does it.

Actually for dir-835-c1 I hope that I have fixed some issues with LEDs and MACs. Anyway, don't have this router as I said, so can't test.

Kind regards,
Seb

### CODE REVIEW

File: target/linux/ath79/dts/ar9344_dlink_dir-825-c1.dts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+			gpios = <&gpio 14 GPIO_ACTIVE_LOW>;
+		};
+
+		orange_wan {
+			label = "d-link:orange:wan";
+			gpios = <&gpio 19 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	leds-2 {
```
Comment (blogic) [inline target/linux/ath79/dts/ar9344_dlink_dir-825-c1.dts]: there is no need to put these into a secondary subnode

File: target/linux/ath79/dts/ar9344_dlink_dir-825-c1.dts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+		blue_wlan5g {
+			label = "d-link:blue:wlan5g";
+			gpios = <&ath9k 0 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy1tpt";
+		};
+	};
+};
+
+&pinmux {
+	free_up_usb_led_gpio: pinmux_free_up_usb_led_gpio {
```
Comment (blogic) [inline target/linux/ath79/dts/ar9344_dlink_dir-825-c1.dts]: use a better name here. look at the naming pattern inside the other DTS files. also the comment can probably be dropped. the directive is self explanatory once correctly named.

File: target/linux/ath79/dts/ar9344_dlink_dir-835-a1.dts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+	leds {
+		compatible = "gpio-leds";
+
+		green_power: green_power {
+			label = "d-link:green:power";
+			gpios = <&gpio 22 GPIO_ACTIVE_LOW>;
+		};
+
+		/* In /etc/board.d/03_gpio_switches GPIO 20 is turned
```
Comment (blogic) [inline target/linux/ath79/dts/ar9344_dlink_dir-835-a1.dts]: dts files should not contain owrt specific info. how about adding the subnode but simply commenting it out ?

File: target/linux/ath79/dts/ar9344_dlink_dir-825-c1.dts (hunk 1 above)
Comment (mkresin) [reply target/linux/ath79/dts/ar9344_dlink_dir-825-c1.dts]: Correction, it make sense to move into an extra node. The ath9k GPIOs are only available after the ath9k driver is loaded. The probe/load of the led gpio led driver would be deferred till this point and we wouldn't have any early boot status indication.

Nevertheless, the node name should be more descriptive. Something like `leds-ath9k` should do the job.

**Comment by blogic** [comment]
On 03/12/2018 11:11, Mathias Kresin wrote:
>
> *@mkresin* commented on this pull request.
>
> ------------------------------------------------------------------------
>
> In target/linux/ath79/dts/ar9344_dlink_dir-825-c1.dts 
> <https://github.com/openwrt/openwrt/pull/1592#discussion_r238207857>:
>
> > +			label = "d-link:blue:wps";
> +			gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
> +		};
> +
> +		orange_power: orange_power {
> +			label = "d-link:orange:power";
> +			gpios = <&gpio 14 GPIO_ACTIVE_LOW>;
> +		};
> +
> +		orange_wan {
> +			label = "d-link:orange:wan";
> +			gpios = <&gpio 19 GPIO_ACTIVE_LOW>;
> +		};
> +	};
> +
> +	leds-2 {
>
> Correction, it make sense to move into an extra node. The ath9k GPIOs 
> are only available after the ath9k driver is loaded. The probe/load of 
> the led gpio led driver would be deferred till this point and we 
> wouldn't have any early boot status indication.
>
> Nevertheless, the node name should be more descriptive. Something like 
> |leds-ath9k| should do the job.
>
makes sense, did not think about that

File: target/linux/ath79/dts/ar9344_dlink_dir-825-c1.dts (hunk 2 above)
Comment (micmac1) [reply target/linux/ath79/dts/ar9344_dlink_dir-825-c1.dts]: I changed it to "enable_gpio_11".

File: target/linux/ath79/dts/ar9344_dlink_dir-835-a1.dts (hunk 3 above)
Comment (micmac1) [reply target/linux/ath79/dts/ar9344_dlink_dir-835-a1.dts]: That actually makes complete sense, I had not thought about this. I removed the comment and added the wan led define. It doesn't hurt to be there, no need to comment it out. Thanks!

**Comment by chunkeey** [comment]
hello, 

I added your patches to my staging.git:

https://git.openwrt.org/?p=openwrt/staging/chunkeey.git;a=commit;h=9c1481f927cf1eea3faa3dbd367781429005e803

**Comment by micmac1** [comment]
Thank you very much!

**Comment by chunkeey** [comment]
Merged. Thanks!

**Pull request closed without merging**